application-collabora-ui/src/main/resources/Collabora/Code/Main.xml

```diff
@@ -197,6 +224,26 @@
     <property>
       <code>require(['jquery'], function($) {
   const collaboraPath ='/rest/collabora/files/';
+  const iframeWindow = document.getElementById('collaboraViewer').contentWindow;
+  var isDocModified = false;
+
+  var receiveMessage = function(event) {
+    // Skip messages that are not from this iframe.
+    if (event.source !== iframeWindow) {
+      return;
+    }
+    var msg = JSON.parse(event.data);
+    if (!msg) {
+      return;
+    }
+
+    if (msg.MessageId == 'App_LoadingStatus' &amp;&amp; msg.Values &amp;&amp; msg.Values.Status == 'Document_Loaded') {
+      // Notify the server that postMessage is going to be used.
```

Comment:
I don't think it's the server that is being notified but rather the JavaScript code in the ``iframeWindow``, i.e. the Collabora editor. Messages are posted between browser windows, they don't leave the browser (client-side) AFAIK.